static/app/views/issueDetails/groupRelatedIssues/index.tsx

```diff
@@ -44,22 +44,27 @@ function GroupRelatedIssues({params}: Props) {
     staleTime: 0,
   });
 
-  const sameRootCauseIssues = relatedIssues?.data
-    .filter(item => item.type === 'same_root_cause')
-    .map(item => item.data);
   // If the group we're looking related issues for shows up in the table,
   // it will trigger a bug in getGroupReprocessingStatus because activites would be empty,
   // thus, we excude it from the list of related issues
-  const groups = sameRootCauseIssues?.filter(id => id.toString() !== groupId)?.join(',');
+  const sameRootCauseIssues = relatedIssues?.data
+    .filter(item => item.type === 'same_root_cause')
+    .map(item => item.data)
+    .filter(id => id.toString() !== groupId)
+    ?.join(',');
+  const traceConnectedIssues = relatedIssues?.data
+    .filter(item => item.type === 'trace_connected')
+    .map(item => item.data)
+    .filter(id => id.toString() !== groupId)
+    ?.join(',');
```

Comment:
I'm not sure how performant this needs to be, but filter and map will iterate over the entire list each time, so we can cut a few of those down with one `.reduce` call:

```typescript
const {sameRootCauseIssues, traceConnectedIssues} = (relatedIssues?.data ?? []).reduce(({mapping}, item) => {
  const issuesList = item.data.filter(id => id.toString() !== groupId)
  if (item.type === 'trace_connected') {
    return {...mapping, traceConnectedIssues: [...traceConnectedIssues, ...issuesList]}
  }
  if (item.type === 'same_root_cause') {
    return {...mapping, sameRootCauseIssues: [...sameRootCauseIssues, ...issuesList]}
  }
}, {sameRootCauseIssues: [], traceConnectedIssues: []})
```

You'll still have to call `.join(',')` after, but I think its an okay tradeoff.

Reply:
Actually it  might be more straight forward if we just retain the keys that come from the data, and change the variable name after:

```typescript
const {same_root_cause: sameRootCauseIssues, trace_connected: traceConnectedIssues} = (relatedIssues?.data ?? []).reduce(({mapping}, item) => {
  const issuesList = item.data.filter(id => id.toString() !== groupId)
  mapping[item.type] = [...(mapping[item.type] ?? []), ...issuesList]
}, {})
``